src/NServiceBus.Persistence.DynamoDB.PersistenceTests/OutboxTests.cs

```diff
@@ -204,4 +236,238 @@ public async Task Should_delete_operations_when_set_as_dispatched_and_more_opera
+        });
+    }
+
+    [Test]
+    public async Task Should_allow_multiple_stores_for_same_id_when_metadata_expired()
+    {
+        var incomingMessageId = Guid.NewGuid().ToString();
+        ContextBag fistAttemptContextBag = configuration.GetContextBagForOutbox();
+
+        var transportOperations = new TransportOperation[]
+        {
+            new(Guid.NewGuid().ToString(), new DispatchProperties(),
+                ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>())
+        };
+
+        using (var transaction = await configuration.OutboxStorage.BeginTransaction(fistAttemptContextBag))
+        {
+            var outboxMessage = new OutboxMessage(incomingMessageId, transportOperations);
+            await configuration.OutboxStorage.Store(outboxMessage, transaction, fistAttemptContextBag);
+
+            await transaction.Commit();
+        }
+
+        await ExpireMetadataRecord(incomingMessageId);
+
+        ContextBag secondAttemptContextBag = configuration.GetContextBagForOutbox();
+        using (var transaction = await configuration.OutboxStorage.BeginTransaction(secondAttemptContextBag))
+        {
+            var outboxMessage = new OutboxMessage(incomingMessageId, transportOperations);
+            await configuration.OutboxStorage.Store(outboxMessage, transaction, secondAttemptContextBag);
+
+            await transaction.Commit();
+        }
+
+        var result = await configuration.OutboxStorage.Get(incomingMessageId, configuration.GetContextBagForOutbox());
+        Assert.AreEqual(transportOperations.Length, result.TransportOperations.Length);
+    }
+
+    [Test]
+    public async Task Should_return_fresh_entry_when_metadata_expired_but_phantom_record_present()
+    {
+        var incomingMessageId = Guid.NewGuid().ToString();
+        ContextBag fistAttemptContextBag = configuration.GetContextBagForOutbox();
+
+        var transportOperations = new TransportOperation[]
+        {
+            new(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>()),
+            new(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>())
+        };
+
+        using (var transaction = await configuration.OutboxStorage.BeginTransaction(fistAttemptContextBag))
+        {
+            var outboxMessage = new OutboxMessage(incomingMessageId, transportOperations);
+            await configuration.OutboxStorage.Store(outboxMessage, transaction, fistAttemptContextBag);
+
+            await transaction.Commit();
+        }
+
+        await ExpireMetadataRecord(incomingMessageId);
+
+        var result = await configuration.OutboxStorage.Get(incomingMessageId, configuration.GetContextBagForOutbox());
+        Assert.That(result, Is.Null);
+    }
+
+    [Test]
+    public async Task Should_return_fresh_entry_when_metadata_expired_but_phantom_records_beyond_query_size_limit_present()
```

Comment:
that seems to be an oddly specific case that feels redundant given that we already expect the result to be null with a phantom record?

Reply:
Similar commenr as below